src/zenml/integrations/s3/flavors/s3_artifact_store_flavor.py

```diff
@@ -101,6 +105,41 @@ def _convert_json_string(
         else:
             raise TypeError(f"{value} is not a json string or a dictionary.")
 
+    @validator("client_kwargs")
+    def _validate_client_kwargs(
+        cls, value: Optional[Dict[str, Any]]
+    ) -> Optional[Dict[str, Any]]:
+        """Validates the `client_kwargs` attribute.
+
+        Args:
+            value: The value to validate.
+
+        Raises:
+            ValueError: If the value is not a valid URL.
+
+        Returns:
+            The validated value.
+        """
+        if value is None:
+            return value
+
+        if "endpoint_url" in value and value["endpoint_url"]:
+            url = value["endpoint_url"].rstrip("/")
+            scheme = re.search("^([a-z0-9]+://)", url)
```

Comment:
instead of extracting the scheme with a custom regex, I would recommend using a standard library like `urllib.parse.urlparse` for this, see https://docs.python.org/3/library/urllib.parse.html
```suggestion
            scheme = urlparse(url).scheme
```